packages/web-components/fast-foundation/src/badge/badge.ts

```diff
@@ -39,21 +18,13 @@ export class Badge extends FoundationElement {
      */
     @attr({ mode: "boolean" })
     public circular: boolean;
-
-    public generateBadgeStyle = () => {
-        if (!this.fill && !this.color) {
-            return;
-        }
-
-        const fill: string = `background-color: var(--badge-fill-${this.fill});`;
-        const color: string = `color: var(--badge-color-${this.color});`;
-
-        if (this.fill && !this.color) {
-            return fill;
-        } else if (this.color && !this.fill) {
-            return color;
-        } else {
-            return `${color} ${fill}`;
-        }
-    };
+    private circularChanged(oldValue: boolean, newValue: boolean): void {
+        DOM.queueUpdate(() => {
+            if (newValue) {
+                this.classList.add("circular");
+            } else {
+                this.classList.remove("circular");
+            }
+        });
```

Comment:
I don't think we need a class here, we can style off the attribute. We've been moving away from classes where we can with checkboxes indeterminate being the exception as there is no attribute there. Also, FYI that If we are going to do work in this scenario, we've updated to make most all of these methods `protected` instead of private to support the extensibility that @scomea calls out.

Reply:
Combining this suggestion with @scomea's, should we simply remove the circular attribute entirely and leave it to css classes or specific implementations? It's always hard with things like this, where "circular" is a very common treatment for this component across design systems, but some DS may _always_ be circular and some may have multiple shape options, and some many not allow circular at all. I think based on this I'm more inclined to remove it and for something like Fluent UI we would probably follow the `appearance` pattern for the allowable shapes. Other DS can very easily do what they need and not have to worry about _not_ being able to remove this option (`Omit` would work I suppose).

Reply:
@nicholasrice Using `classList` was the way I found to _not_ clobber any classes set on the element. The Storybook page actually had some classes defined, but I found those were getting overridden in the previous code. Once I did this I was able to have a class on the custom element and have `circular` get added or removed as needed. Am I missing something?

Reply:
Yeah, calling Element.setAttribute will still clobber the classes hoisted by the element in this case.